src/test/logging_tests.cpp

```diff
@@ -123,15 +139,13 @@ BOOST_FIXTURE_TEST_CASE(logging_LogPrintMacros, LogSetup)
 
 BOOST_FIXTURE_TEST_CASE(logging_LogPrintMacros_CategoryName, LogSetup)
 {
-    // Prevent tests from failing when the line number of the following log calls changes.
-    LogInstance().m_log_sourcelocations = false;
     LogInstance().EnableCategory(BCLog::LogFlags::ALL);
-    const auto concated_categery_names = LogInstance().LogCategoriesString();
+    const auto& concatenated_category_names{LogInstance().LogCategoriesString()};
     std::vector<std::pair<BCLog::LogFlags, std::string>> expected_category_names;
-    const auto category_names = SplitString(concated_categery_names, ',');
+    const auto& category_names{SplitString(concatenated_category_names, ',')};
     for (const auto& category_name : category_names) {
-        BCLog::LogFlags category = BCLog::NONE;
-        const auto trimmed_category_name = TrimString(category_name);
+        BCLog::LogFlags category;
+        const auto& trimmed_category_name{TrimString(category_name)};
         BOOST_TEST(GetLogCategory(category, trimmed_category_name));
         expected_category_names.emplace_back(category, trimmed_category_name);
```

Comment:
Execution will continue past a failed `BOOST_TEST()` if `GetLogCategory()` returns `false` and `category` will remain uninitialized. Thus we will read an uninitialized value in that case. Either restore the initialization to `= BCLog::NONE` or change `BOOST_TEST()` to `BOOST_REQUIRE()`.

Reply:
If BOOST_TEST fails, doesn't the test halt at that point and log a failure?

Reply:
No: https://www.boost.org/doc/libs/1_80_0/libs/test/doc/html/boost_test/testing_tools/tools_assertion_severity_level.html

Reply:
Thanks. Indeed, "reached!" is printed:

```diff
-        BOOST_TEST(GetLogCategory(category, trimmed_category_name));
+        BOOST_TEST(false);
+        BOOST_TEST_MESSAGE("reached!");
```

Reply:
Done